File: Easy/Longest Common Prefix/LongestCommonPrefix.py
# https://leetcode.com/problems/longest-common-prefix


class Solution:

    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        l = 10000000
        small_s = ""
        for s in strs:
            if (len(s) < l):
                l = len(s)
                small_s = s
        for i in range(l):
            c = strs[0][i]
            for s in strs:
                if (c != s[i]):
                    return s[:i]
        return small_s

File: Easy/Longest Common Prefix/LongestCommonPrefix.cpp
// https://leetcode.com/problems/longest-common-prefix

class Solution {
public:
  string longestCommonPrefix(vector<string> &strs) {
    if (strs.empty()) {
      return "";
    }
    string s = *min_element(strs.begin(), strs.end(), [](string a, string b) {
      return a.length() <= b.length();
    });
    string g = "";
    for (int i = 0; i < s.length(); i++) {
      char c = s[i];
      for (int j = 0; j < strs.size(); j++) {
        if (c != strs[j][i]) {
          return g;
        }
      }
      g += c;
    }
    return s;
  }
};